Userland/Libraries/LibJS/Runtime/Intl/RelativeTimeFormat.cpp

```diff
@@ -44,4 +49,79 @@ StringView RelativeTimeFormat::numeric_string() const
+    auto matcher = TRY(get_option(global_object, *options, vm.names.localeMatcher, Value::Type::String, AK::Array { "lookup"sv, "best fit"sv }, "best fit"sv));
+
+    // 5. Set opt.[[LocaleMatcher]] to matcher.
+    opt.locale_matcher = matcher;
+
+    // 6. Let numberingSystem be ? GetOption(options, "numberingSystem", "string", undefined, undefined).
+    auto numbering_system = TRY(get_option(global_object, *options, vm.names.numberingSystem, Value::Type::String, {}, Empty {}));
+
+    // 7. If numberingSystem is not undefined, then
+    if (!numbering_system.is_undefined()) {
+        // a. If numberingSystem does not match the Unicode Locale Identifier type nonterminal, throw a RangeError exception.
+        if (!Unicode::is_type_identifier(numbering_system.as_string().string()))
+            return vm.throw_completion<RangeError>(global_object, ErrorType::OptionIsNotValidValue, numbering_system, "numberingSystem"sv);
+
+        // 8. Set opt.[[nu]] to numberingSystem.
+        opt.nu = numbering_system.as_string().string();
+    }
+
+    // 9. Let localeData be %RelativeTimeFormat%.[[LocaleData]].
+    // 10. Let r be ResolveLocale(%RelativeTimeFormat%.[[AvailableLocales]], requestedLocales, opt, %RelativeTimeFormat%.[[RelevantExtensionKeys]], localeData).
+    auto result = resolve_locale(requested_locales, opt, RelativeTimeFormat::relevant_extension_keys());
+
+    // 11. Let locale be r.[[locale]].
+    auto locale = move(result.locale);
+
+    // 12. Set relativeTimeFormat.[[Locale]] to locale.
+    relative_time_format.set_locale(locale);
+
+    // 13. Set relativeTimeFormat.[[DataLocale]] to r.[[dataLocale]].
+    relative_time_format.set_data_locale(move(result.data_locale));
+
+    // 14. Set relativeTimeFormat.[[NumberingSystem]] to r.[[nu]].
+    if (result.nu.has_value())
+        relative_time_format.set_numbering_system(result.nu.release_value());
+
+    // 15. Let style be ? GetOption(options, "style", "string", « "long", "short", "narrow" », "long").
+    auto style = TRY(get_option(global_object, *options, vm.names.style, Value::Type::String, { "long"sv, "short"sv, "narrow"sv }, "long"sv));
+
+    // 16. Set relativeTimeFormat.[[Style]] to style.
+    relative_time_format.set_style(style.as_string().string());
+
+    // 17. Let numeric be ? GetOption(options, "numeric", "string", « "always", "auto" », "always").
+    auto numeric = TRY(get_option(global_object, *options, vm.names.numeric, Value::Type::String, { "always"sv, "auto"sv }, "always"sv));
+
+    // 18. Set relativeTimeFormat.[[Numeric]] to numeric.
+    relative_time_format.set_numeric(numeric.as_string().string());
+
+    MarkedValueList arguments { vm.heap() };
+    arguments.append(js_string(vm, locale));
+
+    // 19. Let relativeTimeFormat.[[NumberFormat]] be ! Construct(%NumberFormat%, « locale »).
+    auto* number_format = MUST(construct(global_object, *global_object.intl_number_format_constructor(), move(arguments)));
```

Comment:
We should have a template that builds the MVL for us, like `call()` :)

Reply:
oooo now that's an idea